### PR TITLE
Few fixes for PrivateSend

### PR DESCRIPTION
**Pull request by UdjinM6**
Cleaner version of #1307 (rebased, fixed conflicts, squashed 2 commits, removed already merged commit, postponed collateral adjustment for later)

Fixes:
- PS should limit entry size, not mixing amount (_legacy issue_)
- there should be no fee in mixing tx (_legacy issue_)
- make sure `pwalletMain` is not null in `PrepareDenominate` (_shouldn't really happen but just in case_)
- no need for "double" in GetAverageAnonymizedRounds, "float" should be enough

### CODE REVIEW

File: src/darksend.cpp [hunk 1]
```diff
@@ -1672,6 +1672,8 @@ bool CDarksendPool::SubmitDenominate()
 
 bool CDarksendPool::PrepareDenominate(int nMinRounds, int nMaxRounds, std::string& strErrorRet, std::vector<CTxIn>& vecTxInRet, std::vector<CTxOut>& vecTxOutRet)
 {
+    if(!pwalletMain) return false;
+
```
Comment (tgflynn) [inline src/darksend.cpp]: Probably should set strErrorRet here

File: src/darksend.cpp [hunk 2]
```diff
@@ -248,10 +248,10 @@ void CDarksendPool::ProcessMessage(CNode* pfrom, std::string& strCommand, CDataS
                 }
             }
 
-            // Allow lowest denom (at max) as a a fee. Normally shouldn't happen though.
-            // TODO: Or do not allow fees at all?
-            if(nValueIn - nValueOut > vecPrivateSendDenominations.back()) {
-                LogPrintf("DSVIN -- fees are too high! fees: %lld, tx=%s", nValueIn - nValueOut, tx.ToString());
+            // There should be no fee in mixing tx
+            CAmount nFee = nValueIn - nValueOut;
+            if(nFee != 0) {
```
Comment (tgflynn) [inline src/darksend.cpp]: I'm wondering why this change is needed if the clients always create DSVIN's without fees.

Why should the MN care if a fee is included ?  Does it break something ?

Comment (UdjinM6) [inline src/darksend.cpp]: This is just to make sure that client did everything correctly and won't burn funds (un)intentionally. In some _really old_ versions we allowed fees to be payed in the mixing tx which is not the case anymore and all fees are paid via collaterals.

File: src/darksend.cpp (hunk 1 above)
Comment (UdjinM6) [inline src/darksend.cpp]: make sense, will fix 👍
